### PR TITLE
Update `next` branch with changes from `main`

### PR DESCRIPTION
**Pull request by sophschneider**
Updates `next` branch with latest changes from `main`.
1. `git merge main` into a `next` branch
2. Resolve conflicts by manually going through `main` commits and `next` commits to gain context and take what was needed
3. Solved any typescript errors including net new migrated components (e.g. I renamed net new `HorizontalStack`s to `InlineStack`, any net new button props I manually migrated)
4. Consolidated net new se23 styles
5. Made sure storybook ran as expected
6. Made sure doc website ran as expected

> 🚨 Please verify that changes in the last linked commit above look as expected in storybook (see CI for link)

### CODE REVIEW

File: polaris-react/src/components/ActionList/ActionList.tsx
Comment (sophschneider) [inline polaris-react/src/components/ActionList/ActionList.tsx:148]: Had to combine the [ActionList filtering commit](https://github.com/Shopify/polaris/commit/8be227e0c4ac6a03566beb8b9717a721c2f20b07#diff-7b7026efad732c17e4599e3f2036547e7e8fd62c066a59b0648966b5f53fc145R1-R145) with the removal of the custom search field.

cc: @MaxCloutier would you be able to tophat and make sure your changes are merged properly into this feature branch?

File: polaris-react/src/components/ButtonGroup/ButtonGroup.scss
Comment (sophschneider) [inline polaris-react/src/components/ButtonGroup/ButtonGroup.scss:38]: Changes from [this commit](https://github.com/Shopify/polaris/commit/5f94cae98e2a41ca33fae3afce7ad078132609e0#diff-13e95c1f5c388265a5e0623d5ae4f479526d52f179e1e85144a1325f2a8a740d) into main

cc: @mattkubej would you be able to tophat and make sure these changes still work on our feature branch? I think it's just the focus ring on separated buttons right?

File: polaris-react/src/components/Pagination/Pagination.scss
Comment (sophschneider) [inline polaris-react/src/components/Pagination/Pagination.scss:26]: @mattkubej I removed the `se23` when merging this into our feature branch, are the styles still working as expected?

File: polaris-react/src/components/IndexTable/components/Checkbox/Checkbox.scss
Comment (sophschneider) [inline polaris-react/src/components/IndexTable/components/Checkbox/Checkbox.scss:12]: @gwyneplaine @jesstelford I think these styles were missed in the IndexTable cleanup, are they good to remove? The se23 styles seem to just be reseting the styles I deleted above but I haven't tophatted

File: polaris-react/src/components/ButtonGroup/ButtonGroup.scss
Comment (sophschneider) [reply polaris-react/src/components/ButtonGroup/ButtonGroup.scss:38]: I noticed in the [storybook regression tests](https://shopify.chromatic.com/test?appId=5d559397bae39100201eedc1&id=6509f95d4f3050672c787a41) that pressed segmented buttons no longer have a dark right side border from this change. Is that intentional?

|Before|After|
|-|-|
|Screenshot 2023-09-19 at 3 44 45 PM|Screenshot 2023-09-19 at 3 44 58 PM|

File: polaris-tokens/src/themes/base/color.ts
Comment (sophschneider) [inline polaris-tokens/src/themes/base/color.ts:1255]: This change triggered the video thumbnail background colours in the [ui regression tests](https://shopify.chromatic.com/test?appId=5d559397bae39100201eedc1&id=6509f95d4f3050672c787b32), I accepted all as this seems intentional, let me know if it isn't!

cc: @aaronccasanova @lgriffee

File: polaris-react/src/components/ActionList/ActionList.tsx
Comment (sophschneider) [reply polaris-react/src/components/ActionList/ActionList.tsx:148]: @laurkim added a story `WithSearch` on our [feature branch](https://github.com/Shopify/polaris/commit/15447ec357f19be709899ca3bfd7d636631ab5cf#diff-291b2f8b4df06133d317e9eda65b2809cbb37065b8eb0b94d8750db1a598be0c) but now with these changes the search field isn't showing anymore according to our [regression tests](https://shopify.chromatic.com/test?appId=5d559397bae39100201eedc1&id=6509f95d4f3050672c7879ee).

@MaxCloutier is this intentional and we have to update our story or is this a regression?

File: polaris-tokens/src/themes/base/color.ts
Comment (lgriffee) [reply polaris-tokens/src/themes/base/color.ts:1255]: I'll defer to @aaronccasanova and @sarahill on this one since they've been the ones working closely on all the color tokens and migrations.

**Review by lgriffee — APPROVED** [review]
Token changes look good! 🪙✨🎉

File: polaris-tokens/src/themes/base/color.ts
Comment (aaronccasanova) [reply polaris-tokens/src/themes/base/color.ts:1255]: This update is intentional and @sarahill tophatted the original PR's storybook and spinstance.

File: polaris-react/src/components/ActionList/ActionList.tsx
Comment (kyledurand) [reply polaris-react/src/components/ActionList/ActionList.tsx:148]: Max made it opt in so the story that Lo added would need the `allowFiltering` prop. [This PR also brings in a story I added](https://github.com/Shopify/polaris/pull/10572/files#diff-291b2f8b4df06133d317e9eda65b2809cbb37065b8eb0b94d8750db1a598be0cR453) that looks like it does something similar so we should remove one

**Review by kyledurand — APPROVED** [review]
I don't have context on all these changes but the ones I do have look good 👍

**Review by yurm04 — APPROVED** [review]

File: polaris-react/src/components/Filters/Filters.tsx
Comment (sophschneider) [inline polaris-react/src/components/Filters/Filters.tsx:184]: Should match [this commit](https://github.com/Shopify/polaris/commit/2e65a7e98dced44f45df02d15cda78093bac3cf3#diff-f21b98ea8a1f295706a4342f1e9ce543562f6c47d6d0682367e90bb7c23961d8) with a [revert-ish commit](https://github.com/Shopify/polaris/commit/9fed743175079a3967c6591e1e4765f2881181e9). I think it nets in just adding this `useEffect`

cc: @mrcthms could you tophat to make sure that this got merged into our feature branch ok?

File: polaris-react/src/components/ActionList/ActionList.tsx
Comment (sophschneider) [reply polaris-react/src/components/ActionList/ActionList.tsx:148]: Thanks for the context! Removed the `withSearch` story in [this commit](https://github.com/Shopify/polaris/pull/10572/commits/a73a5ac3b378e7695a4df380464f03eac58e1a52)

**Review by mrcthms — APPROVED** [review]
Changes pertaining to Filters look good to me!

**Review by laurkim — APPROVED** [review]
💯 🙌

**Comment by sophschneider** [comment]
I have a few questions about the [UI test changes](https://shopify.chromatic.com/test?appId=5d559397bae39100201eedc1&id=650a53aa9654dfcce4db5116) but I've confirmed they match main so I've accepted them

**Comment by sophschneider** [comment]
@aaronccasanova @lgriffee I've merged in your latest token changes (since you last reviewed Laura), lmk if they look ok!

File: polaris-react/src/components/ButtonGroup/ButtonGroup.scss
Comment (sophschneider) [reply polaris-react/src/components/ButtonGroup/ButtonGroup.scss:38]: I tophatted pagination and segmented buttons and they match the main storybook 👍 we may have to fix the right shadow/border in a follow up PR to main

File: polaris-react/src/components/Pagination/Pagination.scss
Comment (sophschneider) [reply polaris-react/src/components/Pagination/Pagination.scss:26]: I tophatted and it seems to match main 👍

File: polaris-react/src/components/IndexTable/components/Checkbox/Checkbox.scss
Comment (sophschneider) [reply polaris-react/src/components/IndexTable/components/Checkbox/Checkbox.scss:12]: I tophatted against main, it all looks good (the same, no regressions :) )

**Comment by sophschneider** [comment]
I'm going to merge this PR to unblock some folks, if there are any more concerns we can fix with a follow up PR!

**Comment by sophschneider** [comment]
FYI @yurm04 @jesstelford  the modal website examples are looking a bit off when closing the modal. Not sure if this a regression bc the example isn't working on the base branch rn but maybe we should fix with a follow up PR. Might have to do with wrapping the examples in `Frame`

edit: made an issue here https://github.com/Shopify/polaris/issues/10586

Screenshot 2023-09-20 at 10 05 58 AM

File: polaris-react/src/components/Pagination/Pagination.scss
Comment (mattkubej) [reply polaris-react/src/components/Pagination/Pagination.scss:26]: Sorry, just seeing this note. Removing should be good. Thanks!

**Comment by lgriffee** [comment]
> @aaronccasanova @lgriffee I've merged in your latest token changes (since you last reviewed Laura), lmk if they look ok!

Looked at the latest token changes added and they look good!

**Pull request merged**
